### ISSUE TITLE
Spotify SDK dependency

### ISSUE THREAD

**Issue by endanke** [ddddxxx/MusicPlayer#5]
Hi!

I noticed that the iOS support for SDK was blocked by SE-0272, but as far as I know this is now resolved and with Swift 5.3 we can have binary dependencies. Do you have plans to add support for this or would that break backward compatibility? I'm happy to help you out with the changes if Swift 5.3 is supported by your project.

**Comment by ddddxxx** [comment]
Hi, I'm glad you're interested. I managed to create a [wrapper package](https://github.com/ddddxxx/SpotifyiOSWrapper) for SpotifyiOS, but it [breaks macOS build](https://github.com/ddddxxx/MusicPlayer/blob/master/Package.swift#L90). You can try it on master branch with environment variable `LX_ENABLE_SPOTIFYIOS`. Please note [the SpotifyiOS implementation](https://github.com/ddddxxx/MusicPlayer/blob/master/Sources/MusicPlayer/Players/SpotifyiOS.swift) is very old. I just fixed a couple of syntax error, It probably won't work very well (or don't work at all), and I won't get into this anytime soon.

**Comment by endanke** [comment]
I think I've found a solution. In the wrapper package the `SpotifyiOSWrapper` target also need a conditional dependency on `SpotifyiOS` like `.target(name: "SpotifyiOS", condition: .when(platforms: [.iOS]))`. This way it builds on macOS. I assume this is because the dependencies are resolved for all platforms, but the target can still have different dependencies internally.

However, it looks like I have another issue related to code signing and I couldn't find a solution yet. When I tried to build it for an iOS device I get the following error: 
`SpotifyiOS.framework: bundle format unrecognized, invalid, or unsuitable Command CodeSign failed with a nonzero exit code`
I guess either it has to be re-signed or the signing needs to be turned off for that package, but I couldn't get it working yet.

**Comment by endanke** [comment]
I've found a way to fix the code signing error. I'm not sure if this is the right way to do, but I couldn't find any better resources about this so far.
In the original Spotify.framework there's a link to the headers folder and to the binary which are in the /Versions/A subfolder. I noticed that if I move those to the root of the framework and delete the links, then the project builds without a signing issue.

But then it also has an error during runtime which says that an Info.plist is missing from the frameworks inside the xcframework. I've solved that by putting a manually created Info.plist, and then it finally works!

Hopefully this helps. Please check if you know any better ways to do this. If not, I can make a PR with these changes in the wrapper repository.

**Comment by andr-ec** [comment]
@endanke I'd like to see the changes you've made to the wrapper repository. Do you have a branch on your fork?

**Comment by ddddxxx** [comment]
I've updated [SpotifyiOSWrapper](https://github.com/ddddxxx/SpotifyiOSWrapper) to use [official xcframework](https://github.com/spotify/ios-sdk/tree/master/SpotifyiOS.xcframework). I also added [conditional dependency on `SpotifyiOS`](https://github.com/ddddxxx/SpotifyiOSWrapper/blob/master/Package.swift#L21) as @endanke suggested. However I still get `While building for macOS, no library for this platform was found in '[...]SpotifyiOS.xcframework'` error.

For now the Spotify iOS support is still guarded behind `LX_ENABLE_SPOTIFYIOS` flag.
